egs/wsj/s5/steps/combine_ali_dirs.sh

```diff
@@ -1,105 +1,209 @@
+# Verify that phones and trees are compatible in all directories, and than
+# num_jobs files are present, too.
+for src in $@; do
+  if [[ $src != $first_src ]]; then
+    if [[ ! -f $src/num_jobs ]]; then
+      echo "$0: error: Required source file $src/num_jobs is missing."
+      exit 1
+    fi
+    if ! cmp -s $first_src/tree $src/tree; then
+      echo "$0: error: tree $src/tree is either missing or not the" \
+           "same as $first_src/tree."
+      exit 1
+    fi
+    if [[ ! -f $src/phones.txt ]]; then
+      echo "$0: error: Required source file $src/phones.txt is missing."
+      exit 1
+    fi
+    utils/lang/check_phones_compatible.sh $first_src/phones.txt \
+                                          $src/phones.txt || exit 1
+  fi
 done
-sort -m $temp_dir/ali.*.scp > $temp_dir/ali.scp || exit 1;
 
-echo "$0: splitting data to get reference utt2spk for individual ali.JOB.gz files."
-utils/split_data.sh $data $num_jobs || exit 1;
+# All checks passed, ok to prepare directory. Copy model and other files from
+# the first source, as they either checked to be compatible, or we do not care
+# if they are.
+mkdir -p $dest || exit 1
+rm -f $dest/{cmvn_opts,final.mdl,num_jobs,phones.txt,tree}
+$do_ali && rm -f $dest/ali.*.{gz,scp}
+$do_lat && rm -f $dest/lat.*.{gz,scp}
+cp $first_src/{cmvn_opts,final.mdl,phones.txt,tree} $dest/ || exit 1
+cp $first_src/frame_subsampling_factor $dest/ 2>/dev/null  # If present.
+echo $nj > $dest/num_jobs || exit 1
+
+# Make temporary directory, delete on signal, but not on 'exit 1'.
+temp_dir=$(mktemp -d $dest/temp.XXXXXX) || exit 1
+cleanup() { rm -rf "$temp_dir"; }
+trap cleanup HUP INT TERM
+echo "$0: note: Temporary directory $temp_dir will not be deleted in case of" \
+     "script failure, so you could examine it for troubleshooting."
+
+
+# This function may be called twice, once to combine alignments and the second
+# time to combine lattices. The two invocations are as follows:
+#   do_combine ali alignments copy-int-vector $@
+#   do_combine lat lattices   lattice-copy $@
+# where 'ali'/'lat' is a prefix to archive name, 'alignments'/'lattices' go into
+# log messages and logfile names, and 'copy-int-vector'/'lattice-copy' is the
+# program used to copy corresponding objects.
+do_combine() {
+  local ark=$1 entities=$2 copy_program=$3
+  shift 3
+
+  echo "$0: Gathering $entities from each source directory."
+  # Assign all source gzipped archive names to an exported variable, one each
+  # per source directory, so that we can copy archives in a job per source.
+  src_id=0
+  for src in $@; do
+    let src_id++
+    nj_src=$(cat $src/num_jobs) || exit 1
+    # Create and export variable src_arcs_${src_id} for the job runner.
+    # Each numbered variable will contain the list of archives, e. g.:
+    # src_arcs_1="exp/tri3_ali/ali.1.gz exp/tri3_ali/ali.1.gz ..."
+    # ('printf' repeats its format as long as there are more arguments).
```

Comment:
I don't see the -v option for printf on my mac.

Reply:
Should be bash 3.2? I try to verify all stuff on docker with it, checks out here. Could you please try if `printf -v foo 'bar'` works?
```
$ docker run --rm -it bash:3.2
bash-3.2# printf -v foo 'bar'
bash-3.2# echo $foo
bar
```